src/transformers/training_args.py

```diff
@@ -811,7 +811,7 @@ def __post_init__(self):
                 raise ValueError("sharded_ddp is not supported with bf16")
         if (
             is_torch_available()
-            and self.device.type != "cuda"
+            and (self.device.type != "cuda" and self.device.type != "xla")
```

Comment:
Will this be false on TPU? The test is there for that purpose since mixed precision training does not work on TPU.

Reply:
You are right that his won't filter out TPU. We can change it to `not (self.device.type == "xla" and "GPU_NUM_DEVICES" in os.environ)`.